src/views/meeting.js

```diff
@@ -31,24 +34,65 @@ export class Meeting extends React.Component {
         this._onMeetingLeave = this._onMeetingLeave.bind(this);
     }
 
+    /**
+     * Redirects back the home view if no valid meeting url has been detected.
+     *
+     * @inheritdoc
+     */
+    componentDidMount() {
+        if (!this._getMeetingUrl()) {
+            logger.error(
+                'No valid meeting url detected. Params are: ',
+                this.props.location.search
+            );
+            this._onMeetingLeave();
+        }
+    }
+
     /**
      * Implements React's {@link Component#render()}.
      *
      * @inheritdoc
      */
     render() {
-        // TODO: handle the error case where no meeting name is present
+        const meetingUrl = this._getMeetingUrl();
+
+        if (!meetingUrl) {
+            return null;
+        }
 
         return (
             <View name = 'meeting'>
                 <MeetingFrame
                     displayName = { this.props.displayName }
-                    meetingName = { this.props.match.params.name }
+                    meetingUrl = { meetingUrl }
                     onMeetingLeave = { this._onMeetingLeave } />
             </View>
         );
     }
 
+    /**
+     * Parses the current window location for a valid meeting url. The value
+     * {@code null} will be returned if no valid meeting url is found.
+     *
+     * @private
+     * @returns {string|null}
+     */
+    _getMeetingUrl() {
+        const queryParams = new URLSearchParams(this.props.location.search);
+        const meetingUrl = queryParams.get('location');
```

Comment:
after reading the stuff below more closely, this reads a little funny to me.  because we check `isValidMeetingUrl` and if that fails, it seems like it'd be an error but really we just treat it as a meeting name which isn't clear.  i'm thinking we do something like:

```
const location = queryParams.get('location');

if (isValidMeetingUrl(location, VALID_MEETING_HOSTS)) {
  return location;
} else if (isValidMeetingName(location)) {
  return `https://${VALID_MEETING_HOSTS[0]}/${meetingUrl}`;
}

return null;
```